velox/dwio/dwrf/test/CacheInputTest.cpp

```diff
@@ -932,4 +941,19 @@ TEST_F(CacheTest, ssdReadVerification) {
   ASSERT_GT(ioStats_->read().sum(), 0);
   ASSERT_GT(ioStats_->ramHit().sum(), 0);
   ASSERT_GT(ioStats_->ssdRead().sum(), 0);
+
+  // Corrupt SSD cache file.
+  corruptSsdFile(fmt::format("{}/cache0", tempDirectory_->getPath()));
+
+  // Read from the corrupted cache.
+  readData(kSsdBytes);
+  waitForWrite();
+  stats = cache_->refreshStats();
+  // Expect corruptions.
+  ASSERT_GT(stats.numHit, 0);
+  ASSERT_GT(stats.ssdStats->entriesRead, 0);
+  ASSERT_GT(stats.ssdStats->readSsdCorruptions, 0);
+  ASSERT_GT(ioStats_->read().sum(), 0);
+  ASSERT_GT(ioStats_->ramHit().sum(), 0);
+  ASSERT_GT(ioStats_->ssdRead().sum(), 0);
```

Comment:
Can we check if ramHit and ssdRead count don't change? Thanks!

Reply:
Only the last 1/10 of the cache file is corrupted, the reads hitting the first 9/10 would go through successfully.